supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
> We really want the operations to be serialized per containers though, so maybe break down into sub channel per container?

are you actually meaning to extract the container ID from a task (when possible) and add the task to a specific `map[CID]channel`?